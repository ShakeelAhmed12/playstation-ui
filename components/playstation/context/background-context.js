import { createContext, useState } from 'react';

export const BackgroundContext = createContext();

export const BackgroundProvider = ({ children }) => {
    const [activeGame, setActiveGame] = useState('explore');

    return (
        <BackgroundContext.Provider value={{ activeGame, setActiveGame }}>
            {children}
        </BackgroundContext.Provider>
    )
}