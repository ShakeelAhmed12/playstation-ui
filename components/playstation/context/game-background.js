import { createContext, useState } from 'react';

export const BackgroundContext = createContext();

export const GameBackground = ({ children }) => {
    const [activeGame, setActiveGame] = useState('explore');

    return (
        <BackgroundContext.Provider value={{ activeGame, setActiveGame }}>
            {children}
        </BackgroundContext.Provider>
    )
}