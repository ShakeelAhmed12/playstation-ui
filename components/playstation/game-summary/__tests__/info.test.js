import React from 'react';
import renderer from 'react-test-renderer';
import { GameBackground } from '../../context/game-background';

import Info from '../info';

it('renders correctly', () => {
    const tree = renderer
        .create(
            <GameBackground>
                <Info />
            </GameBackground>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});