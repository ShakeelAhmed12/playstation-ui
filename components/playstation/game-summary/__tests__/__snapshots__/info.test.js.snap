// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game info renders correctly 1`] = `
<section
  className="info__StyledInfo-sz20m9-0 kUWuGx"
>
  <span
    className="sub-title"
  >
    Offical news from Playstation | 11/28/2020
  </span>
  <h1>
    Ready for a PS5 Adventure?
  </h1>
  <p>
    ASTRO's PLAYROOM is pre-loaded and waiting for you!
     
    <br />
    explore the new capabilities of PS5 and feel the world...
  </p>
</section>
`;
